src/platform/windows/mod.rs

```diff
@@ -0,0 +1,1482 @@
+{
+    if handle == INVALID_HANDLE_VALUE {
+        return Ok(INVALID_HANDLE_VALUE);
+    }
+
+    unsafe {
+        let mut new_handle: HANDLE = INVALID_HANDLE_VALUE;
+        let ok = kernel32::DuplicateHandle(*CURRENT_PROCESS_HANDLE as HANDLE, handle,
+                                           other_process, &mut new_handle,
+                                           0, winapi::FALSE, flags);
+        if ok == winapi::FALSE {
+            Err(WinError::last("DuplicateHandle"))
+        } else {
+            Ok(new_handle)
+        }
+    }
+}
+
+// duplicate a handle in the current process
+fn dup_handle(handle: HANDLE) -> Result<HANDLE,WinError> {
+    dup_handle_to_process(handle, unsafe { kernel32::GetCurrentProcess() })
+}
+
+// duplicate a handle to the target process
+fn dup_handle_to_process(handle: HANDLE, other_process: HANDLE) -> Result<HANDLE,WinError> {
+    dup_handle_to_process_with_flags(handle, other_process, winapi::DUPLICATE_SAME_ACCESS)
+}
+
+// duplicate a handle to the target process, closing the source handle
+fn move_handle_to_process(handle: HANDLE, other_process: HANDLE) -> Result<HANDLE,WinError> {
+    dup_handle_to_process_with_flags(handle, other_process, winapi::DUPLICATE_CLOSE_SOURCE | winapi::DUPLICATE_SAME_ACCESS)
+}
+
+// get the current process handle as a handle in another process
+#[allow(dead_code)]
+fn current_process_handle_to_process(other_process: HANDLE) -> Result<HANDLE,WinError> {
+    unsafe {
+        let mut new_handle: HANDLE = INVALID_HANDLE_VALUE;
+        let ok = kernel32::DuplicateHandle(kernel32::GetCurrentProcess(), kernel32::GetCurrentProcess(),
+                                           other_process, &mut new_handle,
+                                           winapi::PROCESS_DUP_HANDLE, winapi::FALSE, 0);
+        if ok == winapi::FALSE {
+            Err(WinError::last("DuplicateHandle"))
+        } else {
+            Ok(new_handle)
+        }
+    }
+}
+
+fn create_overlapped() -> Box<winapi::OVERLAPPED> {
+    Box::new(winapi::OVERLAPPED {
+        Internal: 0,
+        InternalHigh: 0,
+        Offset: 0,
+        OffsetHigh: 0,
+        hEvent: 0 as HANDLE
+    })
+}
+
+fn reset_overlapped(ov: &mut winapi::OVERLAPPED) {
+    ov.Internal = 0;
+    ov.InternalHigh = 0;
+    ov.Offset = 0;
+    ov.OffsetHigh = 0;
+    ov.hEvent = 0 as HANDLE;
+}
+
+
+#[derive(Debug, PartialEq)]
+struct WinHandle {
+    h: HANDLE
+}
+
+unsafe impl Send for WinHandle { }
+unsafe impl Sync for WinHandle { }
+
+impl Drop for WinHandle {
+    fn drop(&mut self) {
+        self.close();
+    }
+}
+
+impl Default for WinHandle {
+    fn default() -> WinHandle {
+        WinHandle { h: INVALID_HANDLE_VALUE }
+    }
+}
+
+impl Deref for WinHandle {
+    type Target = HANDLE;
+
+    #[inline]
+    fn deref(&self) -> &HANDLE {
+        &self.h
+    }
+}
+
+impl WinHandle {
+    fn new(h: HANDLE) -> WinHandle {
+        WinHandle { h: h }
+    }
+
+    #[allow(dead_code)]
+    fn invalid() -> WinHandle {
```

Comment:
Well, that's an interesting question.  In this case, at one point I wanted to write WinHandle::invalid() to initialize a struct member.  I changed code so I didn't need to do that any more, but I thought I may need to do that again in the future (same with the earlier dead_code).  So instead of killing the function, I marked it as dead for now, lest I end up writing `WinHandle::new(INVALID_HANDLE_VALUE)` in the future.  Not the worst thing, but that was the rationale.